src/handlers/advertisement.js

```diff
@@ -167,49 +128,60 @@ async function notifyIndexer(cid, peerId) {
   }
 }
 
+let bitsPeerId
+let httpPeerId
+
 async function main(event) {
   try {
-    const peerId = await getPeerId()
+    const { Advertisement, Provider, createExtendedProviderAd } = await import('@web3-storage/ipni') // sry
 
-    // Track the latest read cid and advertisementCid
-    let cid
-    let advertisementCid
+    let headCid = await fetchHeadCid() ?? null
 
-    for (const record of event.Records) {
-      const cidString = record.body
-      cid = CID.parse(cidString)
-
-      const previous = advertisementCid ? { '/': advertisementCid.toString() } : await fetchHeadCid()
-      const addresses = [bitswapPeerMultiaddr]
-
-      // Create the advertisement
-      const rawAdvertisement = {
-        Provider: peerId.toString(),
-        Addresses: addresses,
-        Entries: { '/': cidString },
-        ContextID: { '/': { bytes: Buffer.from(cidString).toString('base64') } },
-        Metadata: metadata,
-        IsRm: false,
-        Signature: await computeAdvertisementSignature(previous, peerId, cid, addresses)
-      }
+    const bits = new Provider({
+      protocol: 'bitswap',
+      addresses: [bitswapPeerMultiaddr],
+      peerId: bitsPeerId ?? await getBitswapPeerId()
+    })
 
-      if (previous) {
-        rawAdvertisement.PreviousID = previous
+    for (const record of event.Records) {
+      let ad
+
+      if (record.body === 'ExtendedProviderHTTP') {
+        const http = new Provider({
+          protocol: 'http',
+          addresses: [httpPeerMultiaddr],
+          peerId: httpPeerId ?? await getHttpPeerId()
```

Comment:
Probs don't need to cache this as it's a 1 time thinger

Reply:
uncached it